astropy/utils/state.py

```diff
@@ -7,6 +7,44 @@
 __all__ = ['ScienceState']
 
 
+class _Context:
```

Comment:
If taking it out, we probably should give it a bit nicer name, like `_ScienceStateContext`.

Reply:
I agree. It could just be `ScienceStateContext` so that people overriding the `set` method of ScienceStates can subclass it more transparently.